# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1219788313 14400
#      Tue Aug 26 18:05:13 2008 -0400
# Node ID 6333da0dfdfde7945fd8bfc13a36789358042027
# Parent  4f1ebb704545e5dba92380e98c020d3fe293eef4
hold.m: if hold is applied to a figure, set state for all child axes objects

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2008-08-26  John W. Eaton  <jwe@octave.org>
+
+	* plot/hold.m: If hold is applied to a figure, set state for all
+	child axes objects.
+
 2008-08-26  Michael Goffioul  <michael.goffioul@gmail.com>
 
 	* plot/backend.m: New function to handle backend switch.
 	* plot/Makefile.in: Add it.
 
 2008-08-26  David Bateman  <dbateman@free.fr>
 
 	* plot/__area__.m: Use __next_line_color__ rather than fixed set
diff --git a/scripts/plot/hold.m b/scripts/plot/hold.m
--- a/scripts/plot/hold.m
+++ b/scripts/plot/hold.m
@@ -39,22 +39,25 @@
 ## @var{h}.
 ## 
 ## @end deftypefn
 
 ## PKG_ADD: mark_as_command hold
 
 function hold (varargin)
 
-  [h, varargin] = __plt_get_axis_arg__ ("hold", varargin{:});
+  if (nargin > 0 && ishandle (varargin{1}))
+    [h, varargin, nargs] = __plt_get_axis_arg__ ("hold", varargin{:});
+  else
+    h = gcf ();
+    nargs = numel (varargin);
+  endif
 
   hold_state = get (h, "nextplot");
 
-  nargs = numel (varargin);
-
   if (nargs == 0)
     if (strcmp (hold_state, "add"))
       hold_state = "replace";
     else
       hold_state = "add";
     endif
   elseif (nargs == 1)
     state = varargin{1};
@@ -66,11 +69,16 @@ function hold (varargin)
       else
 	print_usage ();
       endif
     endif
   else
     print_usage ();
   endif
 
+  if (isfigure (h))
+    axes_objs = findobj (h, "type", "axes");
+    h = [h; axes_objs];
+  endif
+
   set (h, "nextplot", hold_state);
 
 endfunction
